pkg/controllers/deprovisioning/metrics.go

```diff
@@ -25,36 +25,50 @@ func init() {
 	crmetrics.Registry.MustRegister(deprovisioningDurationHistogram)
 	crmetrics.Registry.MustRegister(deprovisioningReplacementNodeInitializedHistogram)
 	crmetrics.Registry.MustRegister(deprovisioningActionsPerformedCounter)
+	crmetrics.Registry.MustRegister(deprovisioningEligibleMachinesGauge)
 }
 
-const deprovisioningSubsystem = "deprovisioning"
-
-var deprovisioningDurationHistogram = prometheus.NewHistogramVec(
-	prometheus.HistogramOpts{
-		Namespace: metrics.Namespace,
-		Subsystem: deprovisioningSubsystem,
-		Name:      "evaluation_duration_seconds",
-		Help:      "Duration of the deprovisioning evaluation process in seconds.",
-		Buckets:   metrics.DurationBuckets(),
-	},
-	[]string{"method"},
+const (
+	deprovisioningSubsystem = "deprovisioning"
+	deprovisionerLabel      = "deprovisioner"
+	actionLabel 			= "action"
 )
 
-var deprovisioningReplacementNodeInitializedHistogram = prometheus.NewHistogram(
-	prometheus.HistogramOpts{
-		Namespace: metrics.Namespace,
-		Subsystem: deprovisioningSubsystem,
-		Name:      "replacement_machine_initialized_seconds",
-		Help:      "Amount of time required for a replacement machine to become initialized.",
-		Buckets:   metrics.DurationBuckets(),
-	})
-
-var deprovisioningActionsPerformedCounter = prometheus.NewCounterVec(
-	prometheus.CounterOpts{
-		Namespace: metrics.Namespace,
-		Subsystem: deprovisioningSubsystem,
-		Name:      "actions_performed",
-		Help:      "Number of deprovisioning actions performed. Labeled by action.",
-	},
-	[]string{"action"},
+var (
+	deprovisioningDurationHistogram = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metrics.Namespace,
+			Subsystem: deprovisioningSubsystem,
+			Name:      "evaluation_duration_seconds",
+			Help:      "Duration of the deprovisioning evaluation process in seconds. Labeled by deprovisioner",
+			Buckets:   metrics.DurationBuckets(),
+		},
+		[]string{deprovisionerLabel},
+	)
+	deprovisioningReplacementNodeInitializedHistogram = prometheus.NewHistogram(
```

Comment:
As an aside: Should this be a histogram or a summary? If it _is_ a histogram, we should probably think about what the buckets should be and if the `DurationBuckets` make sense for all of these deprovisioners to get enough detail.

Reply:
Yeah, I actually think this should be a summary now that I think about it. I think we really only care about specific percentiles, like p50, p90, p99 (although this might not be the full list)

Reply:
The other thing to think about with summary vs. histogram is the ability to aggregate over a set amount of time. I don't know what the right answer is here, but summaries will continue to produce quantiles over the entire lifetime of the pod (you can set `MaxAge` in the opts but, in general, let's assume that the `MaxAge` is infinite) while a histogram can be aggregated, either over the lifetime of the pod or over smaller time periods if desired since all the buckets are stored separately as counters

Reply:
Curious if @jackfrancis if you have any thoughts or experience around the trade-offs between Histograms and Summaries in metrics.

Reply:
I've decided to make both of the metrics that were histograms into summaries here. Based on https://prometheus.io/docs/practices/histograms/#quantiles, the tradeoffs seem to be that if we can get the buckets right, that observing data is a cheaper to compute and also more accurate as quantiles in summaries are estimates.

Reply:
On second thought, after discussion with @jonathan-innis, I haven't heard that our buckets are incorrect or not working for users. If we change these to summaries, it becomes a breaking change, which I'd rather keep it as is to not break users. Going to change it back to histograms.